Library/Homebrew/extend/ARGV.rb

```diff
@@ -268,6 +268,10 @@ def fetch_head?
     include? "--fetch-HEAD"
   end
 
+  def ignore_installed_specs?
+    include? "--ignore-installed-specs"
```

Comment:
I think your three options offer the most flexibility, given what @MikeMcQuaid just added, it would probably be good to add hints to each like "recommended" "not recommended" to make it easier for novices (like me).
```
--build-from-source to build formula against non-stable dependencies (recommended)
--force-bottle to install formula with non-stable dependencie (overrides devel version)
--ignore-installed-specs to use stable specs (not recommended)
```

The fallback behaviour to *stable* seems like a very comfortable approach to me as well.
> ensure users fall back to stable as soon as there's a newer stable than the devel

situation is:
1a godep depends on go
1b go is installed as devel
1c godep is e.g. installed -s as a recommended option
2a asa go has a new version, that version is installed by regular upgrade process (right?)
2b godep is upgraded to the home-brew bottle build (would have to be implemented on your side I guess)

One thing that came ti my mind, how are HEAD versions handled in this situation? Just the same as devel? maybe this is off topic but yeah...